src/GraphQL.Tests/Subscription/SampleObservable.cs

```diff
@@ -0,0 +1,80 @@
+#nullable enable
+
+namespace GraphQL.Tests.Subscription;
+
+internal class SampleObservable<T> : IObservable<T>
+{
+    private readonly List<IObserver<T>> _observers = new List<IObserver<T>>();
+
+    public void Next(T data)
+    {
+        IObserver<T>[] observers;
+        lock (_observers)
+        {
+            observers = _observers.ToArray();
+        }
+        foreach (var observer in observers)
+        {
+            observer.OnNext(data);
+        }
+    }
+
+    public void Error(Exception exception)
+    {
+        IObserver<T>[] observers;
+        lock (_observers)
+        {
+            observers = _observers.ToArray();
+        }
+        foreach (var observer in observers)
+        {
+            observer.OnError(exception);
+        }
+    }
+
+    public void Completed()
+    {
+        IObserver<T>[] observers;
+        lock (_observers)
+        {
+            observers = _observers.ToArray();
+        }
+        foreach (var observer in observers)
+        {
+            observer.OnCompleted();
+        }
+    }
+
+    public IDisposable Subscribe(IObserver<T> observer)
```

Comment:
Just in case it's worth to add null check. The same for `public void Error(Exception exception)`

Reply:
:+1:

Reply:
Done